host/service/systemd/service.go

```diff
@@ -19,10 +19,29 @@ type Service struct {
 	Path string
 }
 
+func isVyOS() bool {
+	if st, err := os.Stat("/config/scripts/"); err != nil || !st.IsDir() {
+		if _, err = os.Stat("/usr/libexec/vyos/init/vyos-router"); err != nil {
+			return true
+		}
+	}
+	return false
+}
+
```

Comment:
Please move this in a VyOS package which just implement the New function and return a systemd Service with the custom Path/Config. The systemd package must remain free of OS specific code.

Reply:
Noted